### PR TITLE
Span logs support

### PR DESCRIPTION
**Pull request by srujann**

### CODE REVIEW

File: src/main/java/com/wavefront/sdk/common/Utils.java [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 import com.wavefront.sdk.entities.tracing.SpanLog;
+import com.wavefront.sdk.entities.tracing.SpanLogsDTO;
 
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.UUID;
 import java.util.regex.Pattern;
 
+import javax.annotation.Nonnull;
```
Comment (hanwavefront) [inline src/main/java/com/wavefront/sdk/common/Utils.java]: use `com.wavefront.sdk.common.annotation.NonNull` instead

File: src/main/java/com/wavefront/sdk/common/Constants.java [hunk 2]
```diff
@@ -80,6 +85,11 @@ private Constants() {
    */
   public static final String COMPONENT_TAG_KEY = "component";
 
+  /**
+   * Tag key for indicating spans log are present for a span.
```
Comment (hanwavefront) [inline src/main/java/com/wavefront/sdk/common/Constants.java]: nit: typo `spans log`

File: src/main/java/com/wavefront/sdk/common/Utils.java [hunk 3]
```diff
@@ -205,6 +211,12 @@ public static String tracingSpanToLineData(String name, long startMillis, long d
         sb.append(sanitize(tag._2));
       }
     }
+    if (spanLogs != null  && !spanLogs.isEmpty()) {
+      sb.append(' ');
+      sb.append(SPAN_LOG_KEY);
```
Comment (hanwavefront) [inline src/main/java/com/wavefront/sdk/common/Utils.java]: needs to be wrapped in quotes (i.e. `"_spanLogs"="true"`), else proxy will reject the span

Comment (vikramraman) [reply src/main/java/com/wavefront/sdk/common/Utils.java]: Wrap in `sanitize` and you should be good.

File: src/main/java/com/wavefront/sdk/direct/ingestion/WavefrontDirectIngestionClient.java [hunk 4]
```diff
@@ -185,6 +195,14 @@ private WavefrontDirectIngestionClient(Builder builder) {
     spansInvalid = sdkMetricsRegistry.newCounter("spans.invalid");
     spansDropped = sdkMetricsRegistry.newCounter("spans.dropped");
     spanReportErrors = sdkMetricsRegistry.newCounter("spans.report.errors");
+
+    sdkMetricsRegistry.newGauge("span_logs.queue.size", spanLogsBuffer::size);
+    sdkMetricsRegistry.newGauge("spans_logs.queue.remaining_capacity",
```
Comment (vikramraman) [inline src/main/java/com/wavefront/sdk/direct/ingestion/WavefrontDirectIngestionClient.java]: typo: `span_logs.queue.`

File: src/main/java/com/wavefront/sdk/common/Utils.java (hunk 3 above)
Comment (srujann) [reply src/main/java/com/wavefront/sdk/common/Utils.java]: fixed

**Pull request merged**